src/nvim/os/env.c

```diff
@@ -76,6 +76,31 @@ char *os_getenvname_at_index(size_t index)
   return name;
 }
 
+/// Check if the environment variable `name` exists and has
+/// a value of either 1, "on", "yes", or "true".
+bool os_env_istruthy(const char *name) FUNC_ATTR_NONNULL_ALL
+{
+  const char *val = os_getenv(name);
+  if (val == NULL || *val == NUL) {  // TODO(Pyrohh): Remove the check for
+    return false;                    // NUL after #2574 is merged.
+  }
+
+  errno = 0;
+  long n = strtol(val, NULL, 10);
```

Comment:
In Windows as in Unix os_getenv() is just getenv(), so I am not expecting any update there.

> I assume it is not possible for os_getenv to return a non-NUL-terminated string

Not normal, but it doesn't seem impossible, even in Unix. putenv() might allow it, or failing that, if a program really wants to place binary data in an environment variable, it might even write directly to **environ. A quick search shows one such case in [SO](http://stackoverflow.com/questions/15767114/binary-enviornment-variables-and-setenv).

Is "1" the only number considered "truthly"? If so can't we just compare against "1\0"?